lib/storage/src/content_manager/consensus/consensus_wal.rs

```diff
@@ -56,8 +56,51 @@ impl ConsensusOpWal {
         high: u64,
         max_size: Option<u64>,
     ) -> raft::Result<Vec<RaftEntry>> {
-        let mut entries = (low..high).map(|id| self.entry(id)).try_collect()?;
-        limit_size(&mut entries, max_size);
+        let mut size = 0;
+
+        let entries = (low..high)
+            .map(|id| self.entry(id))
+            .take_while(|entry| {
+                let entry = match entry {
+                    Ok(entry) => entry,
+                    Err(_) => return true,
+                };
+
+                // It's somewhat unclear how `max_size == Some(0)` should be treated.
+                //
+                // `raft::storage::Storage::entries` documentation and `raft::util::limit_size` implementation
+                // suggest that it means "return only a single entry", but why not request a single entry
+                // using `low`/`high` arguments in that case? 🤔
```

Comment:
Or with just `max_size=1`?

Reply:
Might not work reliably. `max_size` is total size of entries _in bytes_. And Raft log entry may be of arbitrary size, so you'll need some more explicit "marker" for "return (at least) a single entry regardless of its size".